### PR TITLE
Check and skip if post parameter name is empty.

### PR DESCRIPTION
**Pull request by hailin84**
Motivation:
Some weird request bodies start with an '&' character, eg: &name=J&age=17.
In that case, key would be "", will get an exception: java.lang.IllegalArgumentException: Param 'name' must not be empty;
Just check and skip empty key.

Modification:
Check if name is empty before create an Attribute

Result:

Signed-off-by: hailin84 myumen@gmail.com

### CODE REVIEW

**Comment by normanmaurer** [comment]
@fredericBregier PTAL

File: codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java [hunk 1]
```diff
@@ -551,9 +556,14 @@ private void parseBodyAttributes() {
                         ampersandpos = currentpos - 1;
                         String key = decodeAttribute(
                                 undecodedChunk.toString(firstpos, ampersandpos - firstpos, charset), charset);
-                        currentAttribute = factory.createAttribute(request, key);
-                        currentAttribute.setValue(""); // empty
-                        addHttpData(currentAttribute);
+                        // Some wired request body starts with a '&' char, eg: &name=J&age=17.
```
Comment (chrisvest) [inline codec-http/src/main/java/io/netty/handler/codec/http/multipart/HttpPostStandardRequestDecoder.java:559]:
```suggestion
                        // Some weird request bodies starts with an '&' char, eg: &name=J&age=17.
```

**Comment by hailin84** [comment]
Man, is this PR possible to get accepted? We have an issue on our micro service. Plan B is ready of course, but I think that's not a good idea in the long run.

**Comment by normanmaurer** [comment]
@fredericBregier can you have a look please ?

/cc @vietj

**Comment by hailin84** [comment]
I have changed line length. Thanks.

**Comment by hailin84** [comment]
Fixed trailing spaces. Please check again, Thanks.

**Comment by fredericBregier** [comment]
@hailin84 I understand your point, but I'm a bit confused.
Because of a wrong entry, you would like that the decoder skip something wrong, right?
Well, from "protocol" point of view, this is not correct of course.
From a "user/usage" point of view, I understand that having a decoder "more lax" can be a good thing.

Now if the input is the following:

`&=12&name=J&age=17`, then the full element, including value `12` will be ignored/lost.

@hailin84 Could you push again you updated code, such that CI/CD gets a new start?

@normanmaurer WDYT ? Is that OK the decoder being more "lax" according to RFC?

**Review by fredericBregier — COMMENTED** [review]
Not sure 100% but I guess that the following entry will get it wrong:
`&=12&name=J&age=17`
Since `=` will be found just after...
But I agree this is not the case you point out : `&name=J&age=17`

**Comment by hailin84** [comment]
> @hailin84 I understand your point, but I'm a bit confused. Because of a wrong entry, you would like that the decoder skip something wrong, right? Well, from "protocol" point of view, this is not correct of course. From a "user/usage" point of view, I understand that having a decoder "more lax" can be a good thing.
> 
> Now if the input is the following:
> 
> `&=12&name=J&age=17`, then the full element, including value `12` will be ignored/lost.
> 
> @hailin84 Could you push again you updated code, such that CI/CD gets a new start?
> 
> @normanmaurer WDYT ? Is that OK the decoder being more "lax" according to RFC?

I have already pushed code, removed trailing spaces.

**Comment by normanmaurer** [comment]
@fredericBregier I guess we could just add an constructor arg that allows to be less strict ?

**Comment by hailin84** [comment]
> @hailin84 Still some issues
> 
> ```
>   [INFO] Starting audit...
>   Error:  /home/runner/work/netty/netty/codec-http/src/main/java/io/netty/handler/codec/http/multipartHttpPostStandardRequestDecoder.java:435: trailing whitespace [RegexpSingleline]
>   Error:  /home/runner/work/netty/netty/codec-http/src/main/java/io/netty/handler/codec/http/multipartHttpPostStandardRequestDecoder.java:561: trailing whitespace [RegexpSingleline]
>   Audit done.
> ```

Oops, I committed but forget to push. It's OK now. Thanks.

**Comment by hailin84** [comment]
Looks like some unit test cases failed, should I commit modified unit test code?

**Comment by fredericBregier** [comment]
Of course, if Test shall be modified, you have to include them. But take care that tests also serve non-regression, which means a change is only acceptable if it does not change the behavior from previous release from "end user" perspective.
Of course, a Test must be added to show the change is working too (passing the content as required in your example should be ok, contrary to previously).

**Comment by hailin84** [comment]
> Of course, if Test shall be modified, you have to include them. But take care that tests also serve non-regression, which means a change is only acceptable if it does not change the behavior from previous release from "end user" perspective. Of course, a Test must be added to show the change is working too (passing the content as required in your example should be ok, contrary to previously).

The Chinese traditional Spring Festival is coming, I'll check it after the vacation. Thanks.

**Comment by chrishowell** [comment]
Hi @hailin84, I've pushed the tests to https://github.com/netty/netty/pull/12427 - please feel free to merge them into this PR and I'll close mine.

**Pull request closed without merging**